packages/beacon-node/src/chain/seenCache/seenGossipBlockInput.ts

```diff
@@ -427,14 +427,12 @@ export class SeenGossipBlockInput {
         blockCache.block as fulu.SignedBeaconBlock,
         cellsAndProofs
       );
-    } else if (blockCache.cachedData) {
+    } else {
       const firstSidecar = blockCache.cachedData.dataColumnsCache.values().next().value;
       if (!firstSidecar) {
-        return;
+        throw new Error("blockInputCache missing both block and data column sidecar");
       }
       dataColumnSidecars = getDataColumnSidecarsFromColumnSidecar(firstSidecar.dataColumn, cellsAndProofs);
-    } else {
-      throw new Error("blockInputCache missing both block and cachedData");
```

Comment:
I think this still needs to be here for undefined state.  Not sure how this would get to here unless there is a bug in the calling code (no block or cachedData was added somehow) but that code is mostly untested until I finish the refactor with unit tests so lets leave this in for now